mikrotik/resource_interface_veth.go

```diff
@@ -0,0 +1,234 @@
+package mikrotik
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ddelnano/terraform-provider-mikrotik/client"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	tftypes "github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type interfaceVeth struct {
+	client *client.Mikrotik
+}
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &interfaceVeth{}
+	_ resource.ResourceWithConfigure   = &interfaceVeth{}
+	_ resource.ResourceWithImportState = &interfaceVeth{}
+)
+
+// NewInterfaceVethResource is a helper function to simplify the provider implementation.
+func NewInterfaceVethResource() resource.Resource {
+	return &interfaceVeth{}
+
+}
+
+func (i *interfaceVeth) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	i.client = req.ProviderData.(*client.Mikrotik)
+}
+
+// Metadata returns the resource type name.
+func (i *interfaceVeth) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_interface_veth"
+}
+
+// Schema defines the schema for the resource.
+func (i *interfaceVeth) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Creates a Mikrotik interface veth only supported by RouterOS v7+.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Description: "Identifier of this resource assigned by RouterOS",
+			},
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "Name of the interface veth.",
+			},
+			"comment": schema.StringAttribute{
```

Comment:
Currently, `optional+computed` schema fields **without** default value may cause issues, if field is not provided in `resource "..." "..." {}` block. It is happening due to the way our `client` package works.